### ISSUE TITLE
HTML-AAM: where should label be attached to multi-control Edge <input type=file>?

### ISSUE THREAD

**Issue by jasonkiss** [w3c/html-aam#12]
_From @cyns on January 12, 2016 19:50_

Edge uses a textbox and a button for its default rendering of &lt;input type=file&gt;. If the element is wrapped by a &lt;label&gt; or referenced by a &lt;label for&gt; which control (or both) should get the name?

_Copied from original issue: w3c/aria#163_

**Comment by jasonkiss** [comment]
Looks like the default rendering for `input@type=file` is ControlType Group with a LocalizedControlType of "file", and this gets its accessible name from the associated `label`. The Group ControlType has two children: 

1. ControlType Edit with LocalizedControlType "edit", and default name "Enter name of file to upload". But this edit field isn't focusable or editable other than through through the file chooser window that launches upon activating the input.

2. ControlType Button with LocalizedControlType "button", and default name "Browse...". 

Does that sound right to you, @boggydigital ?

**Comment by boggydigital** [comment]
Yes, @jasonkiss is spot on. input id="if" type=file / label for="id" "Author name" would result in the following structure:

ControlType: Group, LocalizedControlType="file", Name="Author name"
+ ControlType: Edit, Name: "Enter name of file to upload" (Note this is localizable name set internally), supports ValuePattern, Value.IsReadOnly = true, support InvokePattern that make file open dialog to show up when invoked
+ControlType: Button, Name: "Browse..." (Note this is localizable name set internally)

Basically the control works same way as other input controls and should require special mapping in my opinion.
